### ISSUE TITLE
Unexpected tiles overlap on a repetitive pattern

### ISSUE THREAD

**Issue by gustavi** [godotengine/godot#54006]
### Godot version

v4.0.dev.20211015.official [f113dc986]

### System information

Windows 10

### Issue description

Tiles overlap on a repetitive pattern:

![image](https://user-images.githubusercontent.com/2736557/137979968-f1b45f4c-ac2e-4911-a1a9-23abd9eca524.png)

### Steps to reproduce

Using the tile in reproduction project (isometric, 100x226 where the "ground zone" is 100x200):

1. Create a `Scene` (`Control ` in my example, you can change the `Scale` to `x=0.5,y=0.5` to get more information on screen)
2. Create a `TileMap`
3. Add a new `TileSet`
4. Change the `Tile Shape` of this `TileSet` to `Isometric`
5. Change the `Tile` of this `TileSet` to `x=200,y=100` (image 1)
6. On `TileSet` add a new Atlas
7. Set the `Texture` with tile in reproduction project (Answer No if Godot ask to auto generate tile)
8. Change `Texture Region Size` to `x=200,y=126` (image 2)
9. On tile properties update the `Texture Offset` to `x=0,y=-26` (image 3)
10. Draw tiles and reproduce the bug (image 4)

![image](https://user-images.githubusercontent.com/2736557/137981155-52d29e05-ff47-4a93-be2f-f4ce0d9470a7.png)

![image](https://user-images.githubusercontent.com/2736557/137981126-d7ea1727-c807-4b1d-8d64-f1713903b8db.png)

![image](https://user-images.githubusercontent.com/2736557/137981455-c3e20ffb-0d59-469d-9eff-cc6c2df688af.png)

![image](https://user-images.githubusercontent.com/2736557/137981670-996de669-e588-401b-b14e-e40d078c3a07.png)

### Minimal reproduction project

[tile_bug.zip](https://github.com/godotengine/godot/files/7376336/tile_bug.zip)

**Comment by groud** [comment]
Yes, but you also need to enable it on the TileMap node itself. In the Node2D/Ordering category.

**Comment by gustavi** [comment]
OK thank you. It was not obvious at all. New tiles editor is really cool but much less intuitive than the old one.
